standard/expressions.md

```diff
@@ -4346,7 +4416,12 @@ equality_expression
     ;
 ```
 
-> *Note*: Lookup for the right operand of the `is` operator must first test as a *type*, then as an *expression* which may span multiple tokens. In the case where the operand is an *expression*, the pattern expression must have precedence at least as high as *shift_expression*. *end note*
+> *Note*: Lookup for the right operand of the `is` operator must first test as a *type*, then as an *expression* which may span multiple tokens. In the case where the operand is an *expression*, the pattern expression must have precedence at least as high as *shift_expression*. *end note*  
+
+<!-- markdownlint-disable MD028 -->
+
+<!-- markdownlint-enable MD028 -->
+> *Note*: There is a grammar ambiguity between *type* and *constant_pattern* in a `relational_expression` on the right-hand-side of `is`; either might be a valid parse of a qualified identifier. In such a case, only if it fails to bind as a type (for compatibility with previous versions of the language), is it resolved to be the first thing found (which must be either a constant or a type). This ambiguity is only present on the right-hand side of such an expression.
```

Comment:
As above how the ambiguity should be resolved is normative so not a Note.

Maybe its been a long day but I’m struggling with “first thing found (which must be either a constant or a type)” – thing? not sure that a good word choice – must? is this "will” or “resolved to first constant or type found by lookup, if any” or something else?

Reply:
If it fails to bind to a type, how can the "first thing found" by a type? I think I'm missing something - an example (not necessarily in the standard) might help me understand what we're talking about here. I agree with Nigel's dislike of "first thing found" but can't suggest anything better until I understand the intent in more detail.